scripts/nbconverted/explore_relationship_AandB.py

```diff
@@ -0,0 +1,585 @@
+row_colors = threshold_groups.map(lut)
+
+sns.clustermap(data_encoded,
+               row_cluster=True,
+               col_cluster=False,
+               metric="correlation",
+               row_colors=row_colors,
+               figsize=(50,50))
+
+
+# ### Collapse gene expression pattern
+# It is difficult to see any trends in large heatmap so collapse the encoded gene expression using mean per group
+
+# In[20]:
+
+
+# Get mean of samples in each group ()
+
+data_encoded_mean_less = pd.DataFrame(data_encoded_labeled[data_encoded_labeled.threshold_group == 'less'].mean(numeric_only=True))
+data_encoded_mean_threshold = pd.DataFrame(data_encoded_labeled[data_encoded_labeled.threshold_group == 'threshold'].mean(numeric_only=True))
+data_encoded_mean_greater = pd.DataFrame(data_encoded_labeled[data_encoded_labeled.threshold_group == 'greater'].mean(numeric_only=True))
+
+
+# In[21]:
+
+
+# Plot mean expression for each group
+plt.figure(figsize=(100, 10))
+sns.heatmap(data_encoded_mean_less.T, annot = True, cmap = "RdBu_r")
+plt.figure(figsize=(100, 10))
+sns.heatmap(data_encoded_mean_threshold.T, annot = True, cmap = "RdBu_r")
+plt.figure(figsize=(100, 10))
+sns.heatmap(data_encoded_mean_greater.T, annot = True, cmap = "RdBu_r")
+
+
+# ### Which features have the largest difference as we cross the threshold?
+
+# In[22]:
+
+
+# Difference in means
+diff_data_encoded = data_encoded_mean_greater - data_encoded_mean_less
+abs_diff_data_encoded = abs(diff_data_encoded)
+
+# Get top 5 features
+top_features = abs_diff_data_encoded[0].nlargest()
+top_features
+
+
+# In[23]:
+
+
+# Check sign of top features
+feature_names = [int(l) for l in top_features.index]
+diff_data_encoded.iloc[feature_names]
+
+
+# In[24]:
+
+
+# What is the weight of the offset vector for this top feature?
+top_feature = top_features.index[2]
+print(offset_vae_space[top_feature])
+
+sns.distplot(offset_vae_space)
+
+
+# ## What is the trend of latent features across the threshold?
+# 
+# Plot trend of encoded values along threshold for top feature
+
+# In[25]:
+
+
+# Join sorted original expression of A and encoded expression of top feature
+original_A_exp = test_samples_sorted[rep_gene_A]
+feature_values = data_encoded[top_feature]
+
+trend_feature_df = pd.merge(original_A_exp.to_frame('gene A untransformed'),
+                      feature_values.to_frame('values of feature {}'.format(top_feature)),
+                      left_index=True, right_index=True)
+trend_feature_df.head()
+
+
+# In[26]:
+
+
+# Plot
+sns.regplot(x='gene A untransformed',
+            y='values of feature {}'.format(top_feature),
+           data = trend_feature_df)
+
+
+# ## Which genes are highly weighted in the feature of significance?
+
+# In[27]:
+
+
+# Read in weight matrix
+weight = pd.read_table(weight_file, header=0, index_col=0).T
+weight.head(5)
+
+
+# In[28]:
+
+
+# Get genes associated with top feature
+top_feature = int(top_features.index[0])
```

Comment:
Is this 106 or 19?

Reply:
See above comment.  Thanks for catching the inconsistency.  There is a lot of manual playing around here that I did to explore the dataset so its a little clunky